packages/amplify_core/test/state_machine/my_state_machine.dart

```diff
@@ -43,7 +43,9 @@ class MyEvent extends StateMachineEvent<MyType> {
   final MyType type;
 
   @override
-  MyPreconditionException? checkPrecondition(MyState currentState) {
+  MyPreconditionException? checkPrecondition(
+    StateMachineState<MyType> currentState,
```

Comment:
If `covariant` is needed for this then I was mistaken and we should add it back.

```suggestion
    MyState currentState,
```

Reply:
It was. I reverted the commit that removed covariant.